depclean-gradle-plugin/src/main/java/se/kth/depclean/analysis/DefaultGradleProjectDependencyAnalyzer.java

```diff
@@ -0,0 +1,331 @@
+package se.kth.depclean.analysis;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import org.codehaus.plexus.component.annotations.Component;
+import org.codehaus.plexus.component.annotations.Requirement;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.ConfigurationContainer;
+import org.gradle.api.artifacts.ResolvedArtifact;
+import org.gradle.api.artifacts.ResolvedDependency;
+import lombok.extern.slf4j.Slf4j;
+import se.kth.depclean.core.analysis.ArtifactTypes;
+import se.kth.depclean.core.analysis.ClassAnalyzer;
+import se.kth.depclean.core.analysis.DefaultClassAnalyzer;
+import se.kth.depclean.core.analysis.DependencyAnalyzer;
+import se.kth.depclean.core.analysis.ProjectDependencyAnalyzerException;
+import se.kth.depclean.core.analysis.asm.ASMDependencyAnalyzer;
+import se.kth.depclean.core.analysis.graph.DefaultCallGraph;
+import se.kth.depclean.DepCleanGradleAction;
+
+/**
+ * This is principal class that perform the dependency analysis in a Gradle project.
+ */
+@Slf4j
+@Component(role = GradleProjectDependencyAnalyzer.class)
+public class DefaultGradleProjectDependencyAnalyzer implements GradleProjectDependencyAnalyzer {
+
+  @Requirement
+  private final ClassAnalyzer classAnalyzer = new DefaultClassAnalyzer();
+
+  @Requirement
+  private final DependencyAnalyzer dependencyAnalyzer = new ASMDependencyAnalyzer();
+
+  /**
+   * If true, the project's classes in target/test-classes are not going to be analyzed.
+   */
+  private final boolean isIgnoredTest;
```

Comment:
Could you create the same parameters that we have in the maven plugins? (could be done in a separate PR)

Reply:
I added all the parameters that we have in the maven plugin except for 3 parameters about which I talked about [here](https://github.com/castor-software/depclean/issues/92#issuecomment-873566633)
Please take a look there and then reply accordingly.

Reply:
I would still implement those parameters, we can have set the default to false for those additional parameters but those are useful.